cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -120,13 +120,25 @@ func newEtcdClient() (client *etcd.Client) {
 			break
 		}
 	}
-	client = etcd.NewClient([]string{*etcd_server})
-	if client == nil {
-		return nil
+	// loop until we have > 0 machines && machines[0] != ""
+	timeout := false
```

Comment:
I would much prefer you to use wait.Poll() here (from /pkg/util/wait/wait.go).  It's our "standard" retry loop.